### PR TITLE
Don't prompt for interactive login on non-interactive terminals

### LINKED ISSUE #1432: `observable deploy` prompts for message in non-interactive shell, throws ERR_TTY_INIT_FAILED
Running `observable deploy` without the `--message` param prompts the user for a deploy message. The prompt is created even when `effects.isTty` is false, causing the process to error with
```
■  Error: TTY initialization failed: uv_tty_init returned EINVAL (invalid argument)
│
└  The full error follows

node:internal/errors:563
    ErrorCaptureStackTrace(err);
    ^

SystemError [ERR_TTY_INIT_FAILED]: TTY initialization failed: uv_tty_init returned EINVAL (invalid argument)
    at new SystemError (node:internal/errors:259:5)
    at new NodeError (node:internal/errors:370:7)
    at new WriteStream (node:tty:93:11)
    at TD.prompt (file:/// ... /node_modules/@clack/core/dist/index.mjs:9:818)
    at Module.te (file:/// ... /node_modules/@clack/prompts/dist/index.mjs:9:7)
    at Module.deploy (file:/// ... /node_modules/@observablehq/framework/dist/deploy.js:238:32)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async file:/// ... /node_modules/@observablehq/framework/dist/bin/observable.js:112:7 {
  code: 'ERR_TTY_INIT_FAILED',
  info: {
    errno: -22,
    code: 'EINVAL',
    message: 'invalid argument',
    syscall: 'uv_tty_init'
  },
  errno: [Getter/Setter],
  syscall: [Getter/Setter]
}
```

To reproduce, run the command with `stdin` closed:
```
observable deploy --debug <&-
```

### PR DESCRIPTION
**Pull request by mythmon**
Fixes #1432

### CODE REVIEW

File: src/deploy.ts [hunk 1]
```diff
@@ -141,6 +141,19 @@ class Deployer {
     }
 
     if (!currentUser) {
+      if (!this.effects.isTty) {
+        if (authError === "unauthenticated" || !apiKey) {
```
Comment (mbostock) [inline src/deploy.ts:145]: Do we need to handle the `"forbidden"` or `null` cases of `authError` here too?

Comment (mbostock) [reply src/deploy.ts:145]: Looks like if `authError` is `null` that implies that `apiKey` is also `null`, so that is already covered. And if `authError` is `"forbidden"`, then the `else` block displays the rejection here. It’s a big hard to follow but it looks correct. 😅

File: src/deploy.ts [hunk 2]
```diff
@@ -141,6 +141,19 @@ class Deployer {
...
+          throw new CliError("No authentication provided");
+        } else {
+          const source =
+            apiKey?.source == "file"
```
Comment (mbostock) [inline src/deploy.ts:149]: `apiKey` can’t be nullish here because we checked `!apiKey` above.

**Comment by mootari** [comment]
Please note that in the case described in #1432 the deploy fails after a successful upload, because the user is prompted for a message (which should likely default to an empty string in that case).

File: src/deploy.ts [hunk 3]
```diff
@@ -141,6 +141,19 @@ class Deployer {
...
 
     if (!currentUser) {
+      if (!this.effects.isTty) {
+        if (authError === "unauthenticated" || !apiKey) {
+          throw new CliError("No authentication provided");
+        } else {
+          const source =
+            apiKey.source == "file"
+              ? ` from ${apiKey.filePath}`
+              : apiKey?.source === "env"
```
Comment (mbostock) [inline src/deploy.ts:151]:
```suggestion
              : apiKey.source === "env"
```

**Review by mbostock — APPROVED** [review]
